lib/v2/luogu/userBlog.js

```diff
@@ -0,0 +1,44 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+const timezone = require('@/utils/timezone');
+
+module.exports = async (ctx) => {
+    const username = ctx.params.username;
+    const rsp = await got.get(`https://www.luogu.com.cn/blog/${username}/_sitemap`);
+    const $ = cheerio.load(rsp.data);
+    const blogTitle = $('title').text();
+    const blogLink = $('.am-alert > a').attr('href');
+
+    const metas = $('.am-u-md-12 .lg-table-big')
+        .map(function () {
+            const a = $(this).find('a').first();
+            return {
+                title: a.text(),
+                link: a.attr('href'),
+            };
+        })
+        .get();
+    const item = await Promise.all(
+        metas.map((meta) =>
+            ctx.cache.tryGet(meta.link, async () => {
+                const rsp = await got.get(meta.link);
+                const $ = cheerio.load(rsp.data);
+                const dateRegex = /(\d{4}[-]\d{2}-\d{2} \d{2}:\d{2}:\d{2})/;
+                const dateMatch = $.text().match(dateRegex);
+                return {
+                    title: meta.title,
+                    link: meta.link,
+                    pubDate: dateMatch ? timezone(parseDate(dateMatch[1]), +8) : new Date().toUTCString(),
```

Comment:
Do not set `pubDate` if there isn't any.
https://docs.rsshub.app/joinus/pub-date.html